src/QsCompiler/Tests.Compiler/TestCases/LinkingTests/SyntaxTreeTrim.qs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+// Trimmer Removes Unused Callables
+namespace Microsoft.Quantum.Testing.SyntaxTreeTrimming {
+    
+    @EntryPoint()
+    operation Main() : Unit {
+        UsedOp();
+    }
+
+    operation UsedOp() : Unit { UsedFunc(); }
+    operation UnusedOp() : Unit { UnusedFunc(); }
+
+    function UsedFunc() : Unit { }
+    function UnusedFunc() : Unit { }
+}
+
+// =================================
+
+// Trimmer Keeps UDTs
+namespace Microsoft.Quantum.Testing.SyntaxTreeTrimming {
+
+    newtype UsedUDT = Int;
+    newtype UnusedUDT = String;
+
+    @EntryPoint()
+    operation Main() : Unit {
+        let x = UsedUDT(3);
+    }
+}
+
+// =================================
+
+// Trimmer Keeps Intrinsics When Told
+namespace Microsoft.Quantum.Testing.SyntaxTreeTrimming {
+
+    @EntryPoint()
+    operation Main() : Unit {
+        UsedIntrinsic();
+    }
+
+    operation UsedIntrinsic() : Unit { body intrinsic; }
+    operation UnusedIntrinsic() : Unit { body intrinsic; }
+}
+
+// =================================
+
+// Trimmer Removes Intrinsics When Told
```

Comment:
I'd suggest to just use the same code snippet above, and basically for all code snippets test them with both keepAllIntrinsics=true and keepAllIntrinsics=false.

Reply:
I think the keepAllIntrinsics sub-feature is orthogonal to the other sub-features of this feature, and I believe unit tests should test orthogonal pieces separately.

As far as duplicating the Q# test case code, it is either duplicate the test case code, or duplicate the testing logic in the unit test. My preference here is to have a separate Q# test case for the two tests, even though the content is identical, because that is the way the testing framework is setup: each unit test consumes its associated test case, and having two unit tests consume and operate off of the same test case is doable, but because it deviates from what the setup expects, it requires more code and work on the unit test side. Long story short, I think it is simpler to have a test case with duplicate content here than to complicate the unit test logic to work two different angles off of the same test case.

Reply:
Ok.